lib/kinu.rb

```diff
@@ -12,6 +12,11 @@ def self.base_uri
     (config.ssl? ? URI::HTTPS : URI::HTTP).build(host: config.host, port: config.port)
   end
 
+  def self.base_upload_uri
+    raise "Kinu.config.upload_host is not set." if config.upload_host.empty?
```

Comment:
In the case of config.host and config.upload_host isn't set, it is raised.
The message could be not suitable. I'll tweak this.